pkg/operator/rules.go

```diff
@@ -63,11 +64,31 @@ func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, labelSelector
 	}, nil
 }
 
-func generateRulesConfiguration(ruleformat RuleConfigurationFormat, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
-	if ruleformat == PrometheusFormat {
-		// Unset partialResponseStrategy field.
-		for i := range promRule.Groups {
+func generateRulesConfiguration(ruleformat RuleConfigurationFormat, version semver.Version, namespace string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
```

Comment:
nit: fewer parameters
```suggestion
func generateRulesConfiguration(ruleformat RuleConfigurationFormat, version semver.Version, promRule monitoringv1.PrometheusRule, logger log.Logger) (string, error) {
```

Reply:
I had tried with that but based on how we call  `generateRulesConfiguration()` it is not straight forward :)

Reply:
Updated